pkg/crypto/crypto.go

```diff
@@ -0,0 +1,104 @@
+package crypto
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/openpgp"
+	"golang.org/x/crypto/openpgp/armor"
+)
+
+// EncryptUsername returns the encrypted value to be used as a username annotation
+func EncryptUsername(username string, key []byte) (string, error) {
+	encryptedUsername, err := encrypt(username, key)
+	if err != nil {
+		return "", err
+	}
+	// Make encrypted data compatible as JSON Patch request body.
+	// Newline characters are marked with text placeholder in order to be easily reversed during decryption
```

Comment:
Does encryption add newline characters? Sorry I am still puzzled by what introduces the newline character. Are you getting it from the ConfigMap data?

Reply:
The encryption itself introduces the newlines. The encrypted data is in the form of an [armor block](https://pkg.go.dev/golang.org/x/crypto/openpgp/armor#Block) with many `\n` characters

Reply:
Please call that out in your comment as it is not immediately obvious. Is `<newline>` is visible in the annotation? Should it be made less obvious? `<wmcomarker>`?